apps/mobile/src/state/protected.ts

```diff
@@ -0,0 +1,87 @@
+import * as SecureStore from 'expo-secure-store';
+import merge from 'lodash.merge';
+import { StateCreator } from 'zustand';
+import { createJSONStorage, persist } from 'zustand/middleware';
+
+import { EntityAdapter, EntitySetter, createEntity } from './entityAdapter';
+import { PROTECTED_KEY, filterObjectKeys } from './utils';
+
+interface SoftwareWallet {
+  id: string;
+  type: 'software';
+  mnemonic: string;
+}
+
+interface HardwareWallet {
+  id: string;
+  type: 'ledger';
+  deviceType: string;
+}
+
+type Wallet = SoftwareWallet | HardwareWallet;
+
+interface ProtectedState {
+  _hasHydrated: boolean;
+  wallet: EntityAdapter<Wallet>;
+}
+
+interface ProtectedAction {
+  setHasHydrated(state: boolean): void;
+}
+
+export type ProtectedStore = ProtectedState & ProtectedAction;
+
+const NOT_PERSIST_KEYS = ['_hasHydrated'];
+
+export const createProtectedStore: StateCreator<
```

Comment:
On naming, maybe secure store is a better name, more aligned with the storage medium?